### DISCUSSION TITLE
Local CDDB

### DISCUSSION THREAD

**Discussion by scruml0rd** [enzo1982/freac#297] [Help & Support]
So I have a database file (FreeDB from 2020) is Fre:ac designed to just query the local file or do I need to host a sever ? I Can query the file with beekeeper and it shows the CD I have in the drive as being in there. Maybe I have the wrong type of file ? Its a ".db" database. Sorry I am new to this.

**Comment by enzo1982** [comment]
I only know of freedb snapshots published as .zip and .tar.bz2 files. To use those, you would unpack them to a disk and point the _Local CDDB path_ in fre:ac at that location. That's the only local CDDB format supported by fre:ac.

I never heard of freedb databases in .db format (which is probably SQLite) before today. There is an alternative CDDB server called [CppCDDB](https://github.com/JoachimSchurig/CppCDDB) that uses an SQLite database and can import freedb's .tar.bz2 files. Maybe your .db file was created with this?

In any case, for using fre:ac's local CDDB feature you'll need the extracted .zip or .tar.bz2 format database. You can still download the last .tar.bz2 snapshot from archive.org at https://web.archive.org/web/20200607003008/http://ftp.freedb.org/pub/freedb/. I think they stopped publishing snapshots in .zip format (which would contain the database in a more Windows-friendly layout) already years before that.

**Reply by Lee-Carre** [reply]
> using fre:ac's local CDDB feature

Awesome; thankyou, this was something I was seeking, too. Especially the pointer to a copy of the last complete revision of FreeDB's dataset, relevant documentation, and sample/reference source-code. +7

I frequently use a laptop, and don't always have InterWeb connectivity.

<1GiB? Less than I was expecting. Good. Seems laughable compared to multiple TiB of storage capacity (I say that while being old enough to remember only too well when 1GiB was a *lot* of data (and a 10GiB HDD was considered large)).

[Parkinson's Law of Data](http://www.catb.org/jargon/html/P/Parkinsons-Law-of-Data.html) for the win. Fellow data-\(whore\|hoarder\)s know what I'm talking about.

Might I suggest adding a pointer (perhaps in a tooltip), either to this GH discussion, else at least a URL to the dataset, within Fre:AC itself. Think of it in terms of enabling discovery of how to achieve what the user seeks.

> SQLite

I would've thought that this would be slower for look-ups, than a simpler/flatter textual file format.

Unless the benefits from compression (or otherwise efficient data packing), with an insignificant performance hit, are worth the added complexity (though, libraries available, of course). An SQL DB is also more easily extensible, if your `SELECT` statements are scoped well (no `SELECT *`, for starters).

> databases in .db format (which is probably SQLite)

In my experience, this is most likely. I've come across quite a few files with that vague extension, which were really SQLite DBs.

> I think they stopped publishing snapshots in .zip format (which would contain the database in a more Windows-friendly layout) already years before that.

Likely; I've seen other projects do the same. Whatever compression algorithm used in .zip files lacks the efficiency needed for large datasets. Other algorithms offer superior ratios, while also being more performant. Sometimes the difference is really stark (hundreds of MiB, depending on the original dataset being compressed).

When hosting, you've gotta store all that extra data, and each transfer eats into your monthly data-transfer quota. Strong incentives to use modern, efficient algorithms.

**Reply by Lee-Carre** [reply]
@enzo1982

> You can still [download the last `.tar.bz2` snapshot from archive](//web.archive.org/web/20200607003008/http://ftp.freedb.org/pub/freedb/)

Hmm, I seem to have trouble downloading much of the file at all; a recurring network error, without the possibility of resuming (resets back to zero bytes). I guess the archive's HTTPd is configured for HTML pages; not downloading several-hundred-MiB files.

I'll try to use wget or similar, once I install+configure it.
In the meantime, do you know of any *other* sources of the database available to fetch for local/offline use?

**Reply by Lee-Carre** [reply]
> In the meantime, do you know of any *other* sources of the database available to fetch for local/offline use?

Trickier than I thought. Either non-existant, or hosted on servers which offer little network capacity (a dozen KiB per second).

I did happen upon a [FreeDB archive from 2020-01-01](//www.archive.org/details/freedb20200111).

**Reply by Lee-Carre** [reply]
It seems that [GnuDB doesn't offer dataset downloads](//searx.tuxcloud.net/search?q=site:gnudb.org+download).

Disappointing.
